Feature/mesh upgrade

selankon opened the pull request.

javierbrk committed:
intial module structure

javierajorge committed:
added batman link information publisher

javierajorge committed:
removed debug prints

javierajorge committed:
consistent renaming

javierajorge committed:
fixed folder name

javierajorge committed:
Added test case

javierajorge committed:
removed unused file

javierajorge committed:
Add batctl dependency

javierbrk committed:
Fixed makefile title

javierbrk committed:
firstdraft

javierajorge committed:
added test case, removed luci dependency

javierajorge committed:
added test case, removed luci dependency

javierbrk committed:
Merge branch 'libremesh:master' into babelnodes

javierajorge committed:
added ubus dependency

javierajorge committed:
added ubus dependency

javierajorge committed:
Merge branch 'babelnodes' of github.com:javierbrk/lime-packages into babelnodes

javierajorge committed:
use random-numgen

javierajorge committed:
use random-numgen

javierbrk committed:
removed hard_ifindex and added new line at the end of file

javierbrk committed:
remove debug print

javierbrk committed:
add link tq parameter and optional best indicator

javierbrk committed:
Merge branch 'master' of github.com:javierbrk/lime-packages into babelnodes

javierbrk committed:
Merge branch 'libremesh:master' into babelnodes

javierbrk committed:
added more tests

javierbrk committed:
added new line at the end of file

javierajorge committed:
package refactoring to include tq index

javierajorge committed:
include mac verification in tests

javierajorge committed:
name refactoring and comments

javierajorge committed:
hook test

javierajorge committed:
added hook execution permision

selankon committed:
Create lime-mesh-upgrade package

javierbrk committed:
removed_files_from_other_commit

javierbrk committed:
Merge remote-tracking branch 'origin/meshwideupdates' into feature/mesh-upgrade

javierbrk committed:
nuevo modelo de datos

javierbrk committed:
wip ...

javierbrk committed:
add uci interface

javierbrk committed:
removed dependencies

javierbrk committed:
testing uci files

javierbrk committed:
fixed underscore typo

javierbrk committed:
test

selankon committed:
Implement set_up_firmware_repository

javierbrk committed:
mesh upgrade lib: improves tests, creates states as constants

javierajorge committed:
save configurations to uci and make tests pass

javierajorge committed:
add inform_download_location and trigger_sheredstate_upgrade

javierbrk committed:
incorporates eupgrade and improves testing
also Adds the following methods:
start_node_download(url)
state()
change_state()

selankon commented on packages/lime-mesh-upgrade/files/usr/lib/lua/lime-mesh-upgrade.lua:
```diff
@@ -0,0 +1,234 @@
+}
+
+--shoud epgrade be disabled ?
+eupgrade.set_workdir("/tmp/mesh_upgrade")
```
We have to implement this function? 

On my local code, before puling changes i havve this code that i was working with:

```lua
function mesh_upgrade.set_workdir(workdir)
    if not utils.file_exists(workdir) then
        os.execute('mkdir -p ' .. workdir)
    end
    if fs.stat(workdir, "type") ~= "dir" then
        error("Can't configure workdir " .. workdir)
    end
    mesh_upgrade.WORKDIR = workdir
    --mesh_upgrade.DOWNLOAD_INFO_CACHE_FILE = mesh_upgrade.WORKDIR .. '/download_status'
    mesh_upgrade.FIRMWARE_LATEST_JSON = mesh_upgrade.WORKDIR .. "/firmware_latest_mesh_wide.json"
    --mesh_upgrade.FIRMWARE_LATEST_JSON_SIGNATURE = mesh_upgrade.FIRMWARE_LATEST_JSON .. '.sig'
end
```

Maybe this code is lost on a push?

selankon replied:
After look a little bit on the code, and blame the file I found that this commit https://github.com/selankon/lime-packages/commit/d7d3d304726daf719cec1a354ffcdd900d1879c9 comments the above function and this one deletes it https://github.com/selankon/lime-packages/commit/b7dd972deaa32cdd5667ab532eaa9152db4d0478.

But the function `mesh_upgrade.set_up_firmware_repository` needs what this code do (`mesh_upgrade.FIRMWARE_LATEST_JSON`): 

https://github.com/selankon/lime-packages/blob/b7dd972deaa32cdd5667ab532eaa9152db4d0478/packages/lime-mesh-upgrade/files/usr/lib/lua/lime-mesh-upgrade.lua#L61

So I supose that is disabled by error, i will add it again on a future commit

javierbrk replied:
I thought about using the eupgrade infrastructure directly, if it that is possible, the code mentioned above is not necessary anymore, 
Also, one of the changes incorporates the report of the downloaded firmware local file url.  
I added a new method to eupgrade to change the repository url. 

But this is just a proposal, We can make a call to discuss this, Maybe I am missing something.

javierbrk replied:
hoo I see, maybe this file can be the same from eupgrade eupgrade.FIRMWARE_LATEST_JSON in this way we dont have to implement the same code here. Just as a suggestion

selankon commented on packages/lime-mesh-upgrade/files/usr/lib/lua/lime-mesh-upgrade.lua:
```diff
@@ -0,0 +1,234 @@
+end
+
+--
+-- {
+--     type= "upgrade",
+--     data={
+--       firmware_ver="xxxx",
+--       repo_url="http://10.13.0.1/lros/api/v1/",
+--       upgrde_state="starting,downloading|ready_for_upgrade|upgrade_scheluded|confirmation_pending|~~confirmed~~|updated|error",
+--       error="CODE",
+--       safe_upgrade_status="",
+--       eup_STATUS="",eup.STATUS_DEFAULT = 'not-initiated' eup.STATUS_DOWNLOADING = 'downloading' eup.STATUS_DOWNLOADED = 'downloaded' eup.STATUS_DOWNLOAD_FAILED = 'download-failed'
+--     },
+--     timestamp=231354654,
+--     id="",
+--     transaction_state="started/aborted/finished",
+--     master_node=""
+-- }
+--
```
commented code

javierbrk replied:
this is just a remainder of the main data type, i found it useful. I think that due to the complexity of this module I can be a good idea to add a readme like other modules have.

selankon commented on packages/shared-state-mesh_wide_upgrade/files/etc/shared-state/hooks/mesh_wide_upgrade/start_upgrade:
```diff
@@ -0,0 +1,42 @@
+			utils.unsafe_shell('logger -p daemon.info -t "mesh upgrade" "mesh upgrade detected upgrade from"' ..
+			s_s_data.data.repo_url .. " " .. node)
+		end
+		utils.unsafe_shell('logger -p daemon.info -t "messwideupgrade" "hoock +++++ "' ..
```
`hoock` maybe is a typo and should be hook?

javierbrk replied:
agree

selankon reviewed (commented):
Just some notes about de development

javierbrk committed:
enables testing with real repo information

javierajorge committed:
ads safeupgrade

javierajorge committed:
Merge branch 'feature/mesh-upgrade' of github.com:javierbrk/lime-packages into feature/mesh-upgrade

selankon committed:
Implement create local repository json data

selankon committed:
Implement share firmware images

javierbrk committed:
Merge branch 'libremesh:master' into feature/mesh-upgrade

javierajorge committed:
adds real unit test and  fix start_node_download

selankon committed:
Add test

javierajorge committed:
Merge branch 'feature/mesh-upgrade' of github.com:selankon/lime-packages into feature/mesh-upgrade

javierajorge committed:
Merge branch 'feature/mesh-upgrade' of github.com:selankon/lime-packages into feature/mesh-upgrade

javierajorge committed:
merge commits and use alternative eupgrade infra
adds new methods to eupgrade like "is_meshupgrade_enabled" and
"set_custom_api_url"
fixes some reduntant methods call

javierajorge committed:
Merge branch 'feature/mesh-upgrade' of github.com:javierbrk/lime-packages into feature/mesh-upgrade

selankon committed:
Ensure local latest json sharing

selankon committed:
Implement become_master_node

selankon committed:
Fix eup to eupgrade

javierajorge committed:
updates the downloaded list of files
adds a destination to share fw files
changes firmware repo
fixes get_repo_base_url()

javierajorge committed:
fixes paths and urls to work with eupgrade
creates links in new latest folder
share_firmware_packages after download has finished successfully
adds a test script for the masternode
improves startupgrade hoock log information

javierajorge committed:
helps network module return the write ip format
defines the boardname file

javierajorge committed:
discards signature check on meshupgrade

selankon committed:
Create shared state acl.d

javierajorge committed:
removes transaction concept

javierajorge committed:
Merge branch 'feature/mesh-upgrade' of github.com:selankon/lime-packages into feature/mesh-upgrade

javierajorge committed:
fixes ready for upgrade simulation for x86

selankon committed:
Refactor set_up_local_repository

selankon committed:
Eliminate divisive language

selankon committed:
Split become_main_node

1. become_main_node
- Start download
- Create latest json
2. get_main_node_status
- Get status of the main node eupgrade process
3. start_firmware_upgrade_transaction
- Linkea los archivos al uhttp
- Comprueba q este todo ok
- llama a inform_download_location de shared state

selankon committed:
Fix main node status function

selankon committed:
Fix get_main_node_status and improve redeability

javierajorge committed:
improves state changes

javierajorge committed:
Merge branch 'feature/mesh-upgrade' of github.com:selankon/lime-packages into feature/mesh-upgrade

javierajorge committed:
fixes start upgrade script and adds some gaurds

selankon committed:
Try to fix mesh upgrade rpcd

javierajorge committed:
lime-mesh-upgrade: enable ubus list and methods call

javierajorge committed:
lime_mesh_upgrade: fixes test failure

selankon committed:
mesh-upgrade: delete unnecessary logs

And other minor improvements about readability.

selankon committed:
mesh-upgrade: add test

javierbrk committed:
lime-mesh-upgrade: fiexes json file name

javierbrk committed:
Merge branch 'feature/mesh-upgrade' from kon

javierbrk committed:
lime-mesh-upgrade: fixes "invalid argument" and file permissions

javierajorge committed:
lime-mesh-upgrade: fixes tests bug related to file permission

javierajorge committed:
lime-mesh-upgrade: improves testing and fixes small issues

javierajorge committed:
lime-mesh-upgrade: removes unused requirements

selankon committed:
mesh-upgrade: refactor to implement get_mesh_upgrade_node_status

javierbrk committed:
lime_mesh_upgrade: adds fwpath

javierbrk committed:
Merges get node status

javierbrk committed:
lime-mesh-upgrade: add eupgrade status and more rpc

selankon committed:
mesh-upgrade: refactor for name consistency

selankon committed:
mesh-upgrade: return eupgrade status on check_eupgrade_download_failed

On this way we can avoid succesive calls to eupgrade.getDownloadStatus

selankon committed:
mesh-upgrade: prevent mesh_upgrade.state nill state

selankon committed:
mesh-upgrade: deprecate STARTING state

And implement main node status

selankon committed:
mesh-upgrade: improve code

selankon committed:
mesh-upgrade: upgrade mesh_upgrade main node initial state

selankon committed:
mesh-upgrade: fix wrong uci cursor

selankon committed:
mesh-upgrade: prevent start mesh wide transaction

 If main node not started

javierbrk committed:
mesh-upgrade: fixes tests and incorporates config persistence

javierbrk committed:
mesh-upgrade: prserve mesh config and set su_timeouts

javierajorge committed:
Merge branch 'feature/mesh-upgrade' of github.com:javierbrk/lime-packages into feature/mesh-upgrade

javierbrk committed:
shared-state-async: implements ubus calls

javierbrk committed:
Shared State Async: adds error handling

javierbrk committed:
shared-state-async: improves format adds simple test

javierbrk committed:
Merge branch 'libremesh:master' into feature/mesh-upgrade

javierbrk committed:
mesh-upgrade: adds retry on error and abort

javierbrk committed:
shared state async: bubble output to user for later processing

javierbrk committed:
shared state async: add new line and fix tests

javierajorge committed:
Merge branch 'feature/mesh-upgrade' of github.com:javierbrk/lime-packages into feature/mesh-upgrade

javierbrk committed:
Merge branch 'libremesh:master' into batmaninfo

javierajorge committed:
shared-state-async: ads bat links info

javierbrk committed:
adds shared state async as aa dependency

javierbrk committed:
Merge branch 'libremesh:master' into babelnodes

javierbrk committed:
shared-state-async: add babel links information support

javierbrk committed:
shared-state-async: add babel links information support

javierbrk committed:
lime-mesh-upgrade: add ipv4 in node status

javierajorge committed:
mesh_upgrade: fixes typos and add tests for confirm

G10h4ck committed:
Merge pull request #1056 from javierbrk/babelnodes

Links information module for shared state according to babeld

G10h4ck committed:
Merge pull request #1055 from javierbrk/batmaninfo

links information according to batman protocol

javierbrk committed:
lime-mesh-upgrade: adds rpc parameters to start_s_u

javierbrk committed:
shared-state-async:errorcode as a number

javierbrk committed:
shared-state-async: updates coment in test file

javierbrk committed:
bat_hosts: fixes rpcd error

javierbrk committed:
shared_state_async: add comment to explain rpcd behavior

G10h4ck committed:
Merge pull request #1094 from javierbrk/fix/ubus_bat_hosts

bat_hosts: fixes rpcd error

G10h4ck committed:
Merge pull request #1086 from javierbrk/shrared-state-ubus

Shrared state Async ubus interface

a-gave committed:
ci: multi-arch-build: enable the build of package index + various fixes

update gh-action-sdk from v5 to v7
prevent that entire workflow fail if a single arch fails
allow to run the workflow manually
disable unnecessary options: default_feed, patches refresh_check, '*.init' files formatting

ilario committed:
Merge pull request #1095 from a-gave/fix/ci/multi-arch-build

ci: multi-arch-build: enable the build of package index + various fixes

selankon committed:
lime-utils: implement DSA support for node status

selankon committed:
lime-utils: add todo

G10h4ck committed:
Remove unmaintained broken packages

Some packages where unmaintaned since so much time that they depends on
packages that doesn't exists anymore on OpenWrt feeds.
Remove those to fix warnings reported by OpenWrt build system.

WARNING: Makefile 'package/feeds/libremesh/bmx6-auto-gw-mode/Makefile' has a dependency on 'bmx6', which does not exist
WARNING: Makefile 'package/feeds/libremesh/lime-proto-bmx6/Makefile' has a dependency on 'bmx6', which does not exist
WARNING: Makefile 'package/feeds/libremesh/lime-proto-bmx6/Makefile' has a dependency on 'bmx6-json', which does not exist
WARNING: Makefile 'package/feeds/libremesh/lime-proto-bmx6/Makefile' has a dependency on 'bmx6-sms', which does not exist
WARNING: Makefile 'package/feeds/libremesh/lime-proto-bmx6/Makefile' has a dependency on 'bmx6-table', which does not exist
WARNING: Makefile 'package/feeds/libremesh/lime-proto-bmx6/Makefile' has a dependency on 'bmx6-uci-config', which does not exist
WARNING: Makefile 'package/feeds/libremesh/lime-webui/Makefile' has a dependency on 'luci-i18n-base-en', which does not exist
WARNING: Makefile 'package/feeds/libremesh/luci-app-batman-adv/Makefile' has a dependency on 'luci-lib-jquery-1-4', which does not exist
WARNING: Makefile 'package/feeds/libremesh/luci-app-batman-adv/Makefile' has a dependency on 'luci-lib-dracula', which does not exist
WARNING: Makefile 'package/feeds/libremesh/luci-app-openairview/Makefile' has a dependency on 'luci-lib-jquery-1-4', which does not exist
WARNING: Makefile 'package/feeds/libremesh/ubus-lime-bmx6/Makefile' has a dependency on 'luci-app-bmx6', which does not exist

G10h4ck committed:
Fix pirania circular dependencies

tmp/.config-package.in:98839:error: recursive dependency detected!
tmp/.config-package.in:98839:	symbol PACKAGE_pirania is selected by PACKAGE_shared-state-pirania
tmp/.config-package.in:59559:	symbol PACKAGE_shared-state-pirania is selected by PACKAGE_pirania

G10h4ck committed:
Merge pull request #1096 from selankon/f/node-status-DSA-support

lime-utils: implement DSA support for node status

G10h4ck committed:
safe-upgrade: fix calibration data partition name

DISCLAIMER: Breaks safe-upgrade retrocompatibility!

Newer OpenWrt rely on calibration data MTD partitions being named 'art'
older OpenWrt on which librerouterOS 1.5 is based, looked for calibration
calibration data for on-SoC radio on 'ART' partition.
In particular librerouter v1 have an on SoC radio which have calibration data
in art partition, safe-upgrade can either support older OpenWrt version or
newer, but not both at least without change the code drastically.
Here in development branch let's support new OpenWrt.

a-gave committed:
ci: multi-arch-build: run one job at a time

javierbrk committed:
Merge branch 'libremesh:master' into feature/mesh-upgrade

javierbrk committed:
mesh-upgrade: error if filenot found

ilario committed:
Merge pull request #1099 from a-gave/ci/multiarch

ci: multi-arch-build: run one job at a time

javierbrk committed:
mesh-upgrade: adds makefile dependencies

a-gave committed:
ci: multi-arch-build: use a different output path

update packages/lime-system/files/etc/uci-defaults/92_add-lime-repos
and README.md accordingly

javierbrk committed:
mesh-upgrade: abort

javierbrk committed:
mesh-upgrade: reports time and abort scheduled upgrade

javierbrk committed:
mesh-upgrade: fixes became main node error

ilario committed:
Merge pull request #1100 from a-gave/ci/multiarch

ci: multi-arch-build: use a different output path

selankon committed:
lime-app: update to v0.2.26

ilario committed:
Merge pull request #1101 from selankon/lime-app-to-version-v0.2.26

lime-app: update to v0.2.26

javierbrk committed:
mesh-upgrade: check image is valid and add timing

a-gave committed:
update changelog

ilario committed:
Merge pull request #1102 from a-gave/changelog

update changelog

javierbrk committed:
adds shared-state rpcd data,error output format

javierbrk committed:
shared-state-ubus: fix licence comments and variable names

javierbrk committed:
mesh-upgrade: fix ipv4 address, add support for sharedstate async

javierbrk committed:
mesh-upgrade: fix makefile dependencies

javierbrk committed:
mesh-upgrade: fixes ip test failure

javierbrk committed:
mesh-upgrade: fixes makefile error

javierbrk committed:
mesh-upgrade:deleted unnecesary file

javierbrk committed:
Merge branch 'libremesh:master' into feature/mesh-upgrade

javierajorge committed:
mesh_upgrade: implements delayed insert

javierbrk committed:
mesh-upgrade: fixes deadlock in publish

javierbrk committed:
mesh-upgrade: cleans log comments

javierbrk committed:
mesh-upgrade: adds readme message

javierajorge committed:
mesh-upgrade: fix ability to start a new upgrade after confirmation

javierbrk committed:
shared-state: remove cached possibly corrupted files from eupgrade

G10h4ck committed:
Merge pull request #1103 from javierbrk/fix/multiwriter_empty_data

Adds shared-state rpcd data,error output format and shared-state-async rpcd reimplementation

G10h4ck committed:
Merge pull request #1098 from G10h4ck/safe_upgrade_art_fixup

safe-upgrade: update calibration data partition name

G10h4ck committed:
Merge pull request #1097 from G10h4ck/packages_clenup

A few cleanups to packages

javierbrk committed:
Merge branch 'libremesh:master' into feature/mesh-upgrade

selankon committed:
mesh-upgrade: create link of publish_mesh_upgrade
